File: RaspberryServer2/src/AppClasses/TaskClass.cpp
#include "TaskClass.h"

TaskClass::TaskClass(string _deviceName, string _buttonName, string _mode){
    this->deviceName = _deviceName;
    this->buttonName = _buttonName;
    this->mode = _mode;
}

TaskClass::TaskClass(){}
TaskClass::~TaskClass(){}

bool TaskClass::createFromJson(string json){
    DynamicJsonBuffer jsonBuffer;
    JsonObject& root = jsonBuffer.parseObject(json);

    //TODO: tirar essa gambs de primeiro usar um char*
    const char* tmp = root["deviceName"];
    const char* tmp2 = root["buttonName"];
    const char* tmp3 = root["mode"];

    if ((tmp == NULL) || (tmp2 == NULL) || (tmp3 == NULL)) {
        return false;
    }
    this->deviceName = string(tmp);
    this->buttonName = string(tmp2);
    this->mode = string(tmp3);

    return true;
}

string TaskClass::parseToJson(){
    DynamicJsonBuffer jsonBuffer;

    JsonObject& root = jsonBuffer.createObject();
    root["deviceName"] = this->deviceName;
    root["buttonName"] = this->buttonName;
    root["mode"] = this->mode;

    string json;
    root.printTo(json);
    return json;
}

string TaskClass::getDeviceName(){
    return this->deviceName;
}
string TaskClass::getButtonName(){
    return this->buttonName;
}
string TaskClass::getMode(){
    return this->mode;
}

void TaskClass::setDeviceName(string _deviceName){
    this->deviceName = _deviceName;
}
void TaskClass::setButtonName(string _buttonName){
    this->buttonName = _buttonName;
}
void TaskClass::setMode(string _mode){
    this->mode = _mode;
}

File: AndroidClient/app/src/main/java/androidclient/automacaoz/raspberry/bruno/azandroidclient/AppClasses/TaskClass.java
package androidclient.automacaoz.raspberry.bruno.azandroidclient.AppClasses;

import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by Bruno on 18/06/2017.
 */

public class TaskClass {
    private final String TAG = "TASK_CLASS";
    String deviceName;
    String buttonName;
    String mode;

    public TaskClass(String deviceName, String buttonName, String mode) {
        this.deviceName = deviceName;
        this.buttonName = buttonName;
        this.mode = mode;
    }

    public TaskClass(String jsonObj){
        JSONObject json;
        try {
            json = new JSONObject(jsonObj);
        } catch (Exception e){
            e.printStackTrace();
            Log.i(TAG, "Erro ao criar o JSON");
            return;
        }


        try {
            this.deviceName = json.getString("deviceName");
            this.buttonName = json.getString("buttonName");
            this.mode = json.getString("mode");
        } catch (JSONException e) {
            e.printStackTrace();
        }

    }

    public String parseToJson(){
        JSONObject json = new JSONObject();
        try {
            json.put("deviceName", deviceName);
            json.put("buttonName", buttonName);
            json.put("mode", mode);
        } catch (JSONException e) {
            e.printStackTrace();
            Log.i(TAG, "Erro ao criar o JSON");
            return null;
        }

        return json.toString();
    }

    public String getDeviceName() {return deviceName;}
    public void setDeviceName(String deviceName) {this.deviceName = deviceName;}
    public String getButtonName() {return buttonName;}
    public void setButtonName(String buttonName) {this.buttonName = buttonName;}
    public String getMode() {return mode;}
    public void setMode(String mode) {this.mode = mode;}
}
